packages/material-ui/src/Icon/Icon.js

```diff
@@ -1,82 +1,104 @@
-  colorSecondary: {
-    color: theme.palette.secondary.main,
-  },
-  /* Styles applied to the root element if `color="action"`. */
-  colorAction: {
-    color: theme.palette.action.active,
-  },
-  /* Styles applied to the root element if `color="error"`. */
-  colorError: {
-    color: theme.palette.error.main,
-  },
-  /* Styles applied to the root element if `color="disabled"`. */
-  colorDisabled: {
-    color: theme.palette.action.disabled,
-  },
-  /* Styles applied to the root element if `fontSize="inherit"`. */
-  fontSizeInherit: {
-    fontSize: 'inherit',
-  },
-  /* Styles applied to the root element if `fontSize="small"`. */
-  fontSizeSmall: {
-    fontSize: theme.typography.pxToRem(20),
-  },
-  /* Styles applied to the root element if `fontSize="large"`. */
-  fontSizeLarge: {
-    fontSize: theme.typography.pxToRem(36),
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...(styleProps.color !== 'inherit' && styles[`color${capitalize(styleProps.color)}`]),
+    ...styles[`fontSize${capitalize(styleProps.fontSize)}`],
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { color, fontSize, classes } = styleProps;
+
+  const slots = {
+    root: [
+      'root',
+      color !== 'inherit' && `color${capitalize(color)}`,
+      `fontSize${capitalize(fontSize)}`,
+    ],
+  };
+
+  return composeClasses(slots, getIconUtilityClass, classes);
+};
+
+const IconRoot = experimentalStyled(
+  'span',
+  {},
+  {
+    name: 'MuiIcon',
+    slot: 'Root',
+    overridesResolver,
   },
-});
+)(({ theme, styleProps }) => ({
+  /* Styles applied to the root element. */
+  userSelect: 'none',
+  width: '1em',
+  height: '1em',
+  // Chrome fix for https://bugs.chromium.org/p/chromium/issues/detail?id=820541
+  // To remove at some point.
+  overflow: 'hidden',
+  display: 'inline-block', // allow overflow hidden to take action
+  textAlign: 'center', // support non-square icon
+  flexShrink: 0,
+  fontSize: {
+    inherit: 'inherit',
+    small: theme.typography.pxToRem(20),
+    medium: theme.typography.pxToRem(24),
+    large: theme.typography.pxToRem(35),
+  }[styleProps.fontSize],
+  // TODO v5 deprecate, v6 remove for sx
+  color: {
+    primary: theme.palette.primary.main,
+    secondary: theme.palette.secondary.main,
+    action: theme.palette.action.active,
+    error: theme.palette.error.main,
+    disabled: theme.palette.action.disabled,
+    inherit: undefined,
+  }[styleProps.color],
+}));
 
-const Icon = React.forwardRef(function Icon(props, ref) {
+const Icon = React.forwardRef(function Icon(inProps, ref) {
+  const props = useThemeProps({ props: inProps, name: 'MuiIcon' });
   const {
     baseClassName = 'material-icons',
-    classes,
     className,
     color = 'inherit',
     component: Component = 'span',
     fontSize = 'medium',
     ...other
   } = props;
 
+  const styleProps = {
+    ...props,
+    fontSize,
```

Comment:
```suggestion
    fontSize,
    baseClassName,
```

for consistency let's add all values that have defaults